File: API/eGYM/Core/DataColumn.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace eGYM
{
    public class DataColumn
    {
        public string PropertyName { get; set; }
        public string LabelDescription { get; set; }
        public DataTypes DataType { get; set; }
        public bool IsFiltered { get; set; }

        public DataColumn(string name, DataTypes type, string description, bool isFiltered = true)
        {
            this.PropertyName = name;
            this.DataType = type;
            this.LabelDescription = description;
            this.IsFiltered = isFiltered;
        }
    }
}

File: UI/e-GYM-WEB/src/app/models/DataColumn.ts
export enum DataTypes {
    Int = 0,
    String = 1,
    Date = 2,
    Boolean = 3,
    Double = 4,
    Currency = 5,
    Time = 6
}

export class DataColumn {
    public PropertyName: string;
    public LabelDescription: string;
    public DataType: DataTypes;
}